cpp/src/copying/copy_range.cu

```diff
@@ -128,18 +127,44 @@ std::unique_ptr<column> copy_range(column_view const& source,
                "Range is out of bounds.");
   CUDF_EXPECTS(target.type() == source.type(), "Data type mismatch.");
 
-  auto p_ret = std::make_unique<column>(target, stream, mr);
-  if ((!p_ret->nullable()) && source.has_nulls()) {
-    p_ret->set_null_mask(
-      create_null_mask(p_ret->size(), ALL_VALID, stream, mr), 0);
+  if (cudf::is_fixed_width(target.type())) {
```

Comment:
This should be done via specialization of a type dispatched function object, not runtime checks like this. Otherwise it will become intractable as we add more types in the future.